### ISSUE TITLE
project.views.project_auth magic numbers

### ISSUE THREAD

**Issue by bemoody** [MIT-LCP/physionet-build#350]
To quote the documentation of project.views.project_auth:

    """
    Authorization decorator for project. Also adds project information
    to kwargs.
    
    auth_mode is one of the following:
    - 0 : the user must be an author.
    - 1 : the user must be the submitting author.
    - 2 : the user must be an author or an admin
    
    post_auth_mode is one of the following and applies only to post:
    - 0 : no additional check
    - 1 : the user must be an author, and the project must be in one of
          its author editable stages.
    - 2 : the user must be the submitting author, and the project must
          be in one of its author editable stages.
    """

These magic numbers are hard to understand...

- auth_mode=1 is more restrictive than auth_mode=0 which is more restrictive than auth_mode=2.

- post_auth_mode magic numbers bear no relation to auth_mode magic numbers.

- It's not obvious to me that auth_mode=1 (read access for submitting author only) is something we'd ever want to use.  (It's only used by 'move_author', which is POST-only.)

- "admin" is probably not the right level of granularity (we may want to have non-"admin" reviewers who have the ability to view certain projects... not that we need to implement such a capability right now, but it'd be better for the interface and documentation to reflect that future possibility.)

Perhaps replace these magic numbers with enumerations, or define the permissions via more explicit keywords ('allow_editor=True' instead of 'auth_mode=2' ?)

**Comment by cx1111** [comment]
Good idea. I initially came up with decorators to reduce boiler code but this is definitely confusing.

Also class based views may potentially solve this (new reading for me too): https://docs.djangoproject.com/en/2.1/topics/class-based-views/

**Comment by bemoody** [comment]
Another issue is that project.views.edit_metadata_item is currently
*not* using project_auth, because that view has more complicated
restrictions (accessible to the assigned editor, but only during the
copyediting stage.)

It would be better to incorporate those restrictions into project_auth.
